consensus/propagation/commitment.go

```diff
@@ -124,13 +121,19 @@ func chunkToPartMetaData(chunk *bits.BitArray, partSet *proptypes.CombinedPartSe
 // time a proposal is received from a peer or when a proposal is created. If the
 // proposal is new, it will be stored and broadcast to the relevant peers.
 func (blockProp *Reactor) handleCompactBlock(cb *proptypes.CompactBlock, peer p2p.ID, proposer bool) {
+	err := blockProp.validateCompactBlock(cb)
+	if !proposer && err != nil {
```

Comment:
this is not validating the compact block when we're the proposer. if we enable this, some chaos start happening for some reason.

Reply:
Also, the new proposal is not verified in the consensus reactor. So it's fine not to verify it here too

Reply:
[not blocking]
maybe only validate the proposal if we're not the proposer to avoid the error?

Reply:
yes, that's how it's currently done